src/transformStateWithClones.js

```diff
@@ -8,6 +8,23 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+    for (const key in actions) {
+    switch (actions[key].type) {
+      case 'addProperties':
+        for (const extraKey in actions[key].extraData) {
+          state[extraKey] = actions[key].extraData[extraKey];
```

Comment:
Ensure that you are creating a new state object before modifying it. Directly modifying `state` here will affect the original object, which is not desired.